Documentation/proposal-runconfig_inside_osv.md

```diff
@@ -1,55 +1,286 @@
+"dummy" IP and service from where the guest OS can get the information from the cloud-init file.
+
+However, if we are to support a local environment using the same cloud-init approach, it needs to
+get the data somehow. Since Captan invokes QEMU, the only viable option is to attach a second volume
+to it.
+
+Although attaching the second volume locally might be done using `file2img` script from
+[this](https://github.com/cloudius-systems/osv/commit/56c6195) commit, there is no need for this.
+Capstan is able to set the boot command without the cloud-init support. So, to run image locally, one
+would use:
+```
+capstan run --execute "agent run <name-of-config> [<key-value>, <key-value>, ...]"
+
+Example:
+capstan run --execute "agent run pizza IP=192.168.0.55 THE_OUTPUT_DIR=/etc/out MONGO_DB_URL=my.mongo.local"
+```
+Note that --execute argument is exactly the same as the bootcmd int the cloud-init.
 
 
 ### Accessing description of a composed image
+When user downloads an already composed OSv image, she is not aware of what configuration names are
+used and what parameters are required for them. It must be possible to obtain this information from
+the image.
 
-It will be possible to print runtime options of a composed image using capstan:
+I propose the following solution:
+
+### Result
+
+OSv Agent should support some additional commands. `list` command would print all available named
+configurations into OSv console:
+```
+$ agent list
+
+Available configuration names:
+burek
+pizza
 ```
-$ capstan runtime info mike/my-app
+Then `describe` command would print detailed information about the provided configuration name.
 ```
-This command will print all the available config_set names and details only by using data from the image itself (as opposed to reading meta/run.yaml from the current directory on the host).
+$ agent describe <name-of-config>
+```
+With the command above user would get enough information to be able to provide complete boot command.
+
+Commands `agent list` and `agent describe` could be set as boot command. This way user would be able
+to use them directly on OpenStack. But ofcorse, we can simplify things for local environment by
+implementing additional command for Capstan tool that would be setting the two boot commands for us
+and only showed us result.
+
+### Implementation details
+Printing list of named configurations is as simple as listing all filenames from `/osv-runtime`
+folder.
 
-#### Proposed solution
-While composing, Capstan could store the original meta/run.yaml file into OSv image. Then the `capstan runtime info` command could access this file and print it.
+But in OSv image there is currently no documentation data about named configurations stored. We could
+fix this by upgrading Capstan: on COMPOSE time it would not only generate `/osv-runtime` files, but
+also corresponding `/osv-runtime-docs` files. OSv agent would then simply read and print appropriate
```

Comment:
What if we make use of some `/usr/bin` for runtimes and `/usr/man` for the docs?